nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/UpdateBoxFileMetadataInstance.java

```diff
@@ -0,0 +1,338 @@
+                                         final String fileId) {
+        try {
+            final Metadata metadata = boxFile.getMetadata(templateScope, templateName);
+            getLogger().debug("Retrieved existing metadata for file {}", fileId);
+            return metadata;
+        } catch (final BoxAPIResponseException e) {
+            if (e.getResponseCode() == 404) {
+                getLogger().info("No existing metadata found for file {} with template {}. Creating new metadata instance.",
+                        fileId, templateName);
+                return new Metadata(templateScope, templateName);
+            }
+            throw e;
+        }
+    }
+
+
+    private Set<String> updateMetadata(final Metadata metadata,
+                                       final Map<String, Object> desiredState) {
+        final Set<String> processedKeys = new HashSet<>();
+        final List<String> currentKeys = metadata.getPropertyPaths();
+
+        // Remove fields not in desired state
+        for (final String propertyPath : currentKeys) {
+            final String fieldName = propertyPath.substring(1); // Remove leading '/'
+
+            if (!desiredState.containsKey(fieldName)) {
+                metadata.remove(propertyPath);
+                processedKeys.add(fieldName);
+                getLogger().debug("Removing metadata field: {}", fieldName);
+            }
+        }
+
+        // Add or update fields
+        for (final Map.Entry<String, Object> entry : desiredState.entrySet()) {
+            final String fieldName = entry.getKey();
+            final Object value = entry.getValue();
+            final String propertyPath = "/" + fieldName;
+
+            if (updateField(metadata, propertyPath, value, currentKeys.contains(propertyPath))) {
+                processedKeys.add(fieldName);
+            }
+        }
+
+        return processedKeys;
+    }
+
+    private boolean updateField(final Metadata metadata,
+                                final String propertyPath,
+                                final Object value,
+                                final boolean exists) {
+
+        final String stringValue = value != null ? value.toString() : null;
+        if (exists) {
+            final Object currentValue = metadata.getValue(propertyPath);
+            final String currentStringValue = currentValue != null ? currentValue.toString() : null;
+
+            // Only update if values are different
+            if (Objects.equals(currentStringValue, stringValue)) {
+                return false;
+            }
+
+            // Update with appropriate type
+            if (value instanceof Number) {
+                metadata.replace(propertyPath, ((Number) value).doubleValue());
+            } else if (value instanceof List) {
+                metadata.replace(propertyPath, convertListToStringList((List<?>) value));
+            } else {
+                metadata.replace(propertyPath, stringValue);
+            }
```

Comment:
Nit: we can use pattern matching.
```suggestion
            // Update with appropriate type
            switch (value) {
                case Number n -> metadata.replace(propertyPath, n.doubleValue());
                case List<?> l -> metadata.replace(propertyPath, convertListToStringList(l));
                default -> metadata.replace(propertyPath, stringValue);
            };
```